Ginger/Ginger/AutomatePageLib/AddActionMenu/WindowExplorer/Common/WindowExplorerPage.xaml.cs

```diff
@@ -224,254 +355,190 @@ public void ShowAsWindow(eWindowShowStyle windowStyle = eWindowShowStyle.Free)
+                //ActionsTab.IsSelected = true;
+            }
             else
             {
-                StepstoUpdateActionRow.Height = new GridLength(0);
+                //StepstoUpdateActionRow.Height = new GridLength(0);
             }
-            GingerCore.General.LoadGenericWindow(ref _GenWin, null, windowStyle, Title, this, closeEventHandler: CloseWindow);            
+            GingerCore.General.LoadGenericWindow(ref _GenWin, null, windowStyle, Title, this, closeEventHandler: CloseWindow);
         }
 
         private async void CloseWindow(object sender, EventArgs e)
         {
             //stop Live Spy
-            if (LiveSpyButton.IsChecked == true)
-                if (dispatcherTimer != null)
-                    dispatcherTimer.IsEnabled = false;
+            //if (LiveSpyButton.IsChecked == true)
+            //    if (dispatcherTimer != null)
+            //        dispatcherTimer.IsEnabled = false;
 
             //stop Recording
-            if (RecordingButton.IsChecked == true)
-                StopRecording();
+            //if (RecordingButton.IsChecked == true)
+            //    StopRecording();
 
-            if(WindowControlsTreeView.IsSearchRunning())
+            if (xWindowControlsTreeView.IsSearchRunning())
             {
-                await WindowControlsTreeView.CancelSearchAsync();
+                await xWindowControlsTreeView.CancelSearchAsync();
             }
 
             _GenWin.Close();
         }
 
         void InitTree(ITreeViewItem RootItem)
         {
-            WindowControlsTreeView.Tree.ClearTreeItems();
+            xWindowControlsTreeView.Tree.ClearTreeItems();
             mRootItem = RootItem;
 
-            mTreeRootItem = WindowControlsTreeView.Tree.AddItem(RootItem);
-            mTreeRootItem.IsExpanded = false;            
+            mTreeRootItem = xWindowControlsTreeView.Tree.AddItem(RootItem);
+            mTreeRootItem.IsExpanded = false;
         }
-        
-        private void RefreshTreeControls()
+
+        private async Task RefreshPageSrcContent()
         {
-            WindowControlsTreeView.Tree.ClearTreeItems();
-            if (WindowsComboBox.SelectedValue != null && mRootItem != null)
+            //xHTMLTree.Items.Add(new PageSrcParser(((SeleniumDriver)mContext.Agent.Driver).GetPageHTML())
+            if (mWindowExplorerDriver is GenericAppiumDriver)
             {
-            TreeViewItem TVI = WindowControlsTreeView.Tree.AddItem(mRootItem);
-            TVI.IsExpanded = false;
-        }
-        }
-        
-        private void LiveSpyHandler(object sender, RoutedEventArgs e)
-        {
-            //stop Recording
-            if (RecordingButton.IsChecked == true)
-                StopRecording();
-
-            if (WindowsComboBox.SelectedValue != null)
+                xXMLPageSrcViewer.Visibility = Visibility.Visible;
+                string pageSrcXML = await((GenericAppiumDriver)mWindowExplorerDriver).GetPageSource();
```

Comment:
why specific for AppiumDriver? need to use interface

Reply:
**same as above/previous comment**